Khan/wonder-blocks
# Update ModalLauncher to use ReactDOM.createPortal

This is the first step in support tooltips in webapp.  We need both modal and tooltip to be using React 16 portals.  This PR update ModalLauncher to use this new API.  For portals hosted within the modal such as the dropdown portal, we use the modal backdrop as the container so that the dropdown isn't clipped by the modal dialog.

Test Plan:
- load http://localhost:6060/#!/MultiSelect
- open the modal
- click on the multi-select opener, see that the dropdown isn't clipped by the modal dialog

<img width="1279" alt="screen shot 2018-08-14 at 4 41 04 pm" src="https://user-images.githubusercontent.com/1044413/44118531-7bbfa2fe-9fe4-11e8-86a7-7e2ba4cdfff3.png">
